logstash-core-event/lib/logstash/event.rb

```diff
@@ -98,7 +98,13 @@ def clone
       copy[k] = begin v.clone rescue v end
     end
 
-    self.class.new(copy)
+    cloned = self.class.new(copy)
+    
+    # Clone metadata too
+    @metadata.each do |k,v|
+      cloned["@metadata"][k] = begin v.clone rescue v end
+    end
+    cloned
```

Comment:
@ph Sorry to barge in, but I'm interested in this for the clone filter, to my understanding, either you need to loop in the hash (and nested hash etc)  or you can use a Marshal "hack" 
`copied_data = Marshal.load(Marshal.dump(complex_data))`
Do you think it is usable performance wise ?
Reference => http://thingsaaronmade.com/blog/ruby-shallow-copy-surprise.html